scripts/kconfig/kconfiglib.py

```diff
@@ -4751,10 +4751,15 @@ def __str__(self):
         defined in multiple locations will return a string with all
         definitions.
 
+        Symbols without menu nodes return the repr() result.
+
         The returned string does not end in a newline. An empty string is
         returned for undefined and constant symbols.
         """
-        return self.custom_str(standard_sc_expr_str)
+        if len(self.nodes) == 0:
+            return repr(self)
```

Comment:
exactly why is this needed ?

I don't see from the commit message a justification for this change to Kconfiglib, and as already mentioned we should be most careful in not deviating from the kconfig language specification, although extensions can be acceptable.
Of course they must always be well justified.

Have side-effects of this change be investigated ?
I could imagine other parts of kconfiglib actually uses the returned empty string for something.

Reply:
thanks for the explanation.

Feel free to keep it for debuging, but unless there is a good reason for this at the end then I prefer that it be dropped before merging. Let's minimize our customization to kconfiglib.

Marking PR `DNM` for now.

Reply:
> I need it for debugging until then.

I mean... isn't this what `git stash` is for?